frontend/libs/studio-components/src/components/StudioBlobDownloader/StudioBlobDownloader.test.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { StudioBlobDownloader } from './StudioBlobDownloader';
+import type { StudioBlobDownloaderProps } from './StudioBlobDownloader';
+
+describe('StudioBlobDownloader', () => {
+  type ExampleData = {
+    testField1: string;
+    testField2: number;
+  };
+
+  const mockData: ExampleData = {
+    testField1: 'test',
+    testField2: 1,
+  };
+
+  global.URL.createObjectURL = jest.fn();
+  global.URL.revokeObjectURL = jest.fn();
```

Comment:
What do you think about mocking `blobDownloader.handleDownloadClick();` instead? It seems like this component is relying on implementation details. I believe it would be better to depend on the `BlobDownloader` interface instead. Any thoughts about that? 😊

Reply:
The reason we want to avoid tight dependencies on implementation details is that it allows us to use dedicated classes or interfaces instead. This enables us to change the implementation of, for example, BlobDownloader in the future without having to modify the tests for this component. The component is only concerned with the interface of BlobDownloader, not its implementation.